Static content for new auth page (#553)
department-of-veterans-affairs/developer-portal: src/content/apiDocs/oauthTechnical.mdx

  '{ CodeWrapper }': ../../components
---

This page walks you through the steps toward authorization and authentication for using OAuth APIs on the Lighthouse platform. We provide examples throughout this document that you can copy and use to get started. 

They are shown with generic cURL commands. Select your API from the drop-down near each cURL command to populate the examples with your API-specific URLs, parameters, and scopes.

## On this Page: {tabIndex="-1"}

- [Getting Started, Initiating the Authorization Code Flow](#getting-started)
- [Building OpenID Connect Applications](#building-openid-connect-applications)
- [Scopes](#scopes)
- [ID Token](#id-token)
- [Test Users](#test-users)

### It’s also good to know that: 
- The access credentials we supply are for the sandbox environment only and will not work in the production environment. 
- This page provides examples that show authorization server URLs in the sandbox environment, which will differ depending on the API. 
- When your application is ready, you may [apply for production access.](https://developer.va.gov/go-live)

[Return to top](#on-this-page)
## Getting Started {tabIndex="-1"}

> ---

Author: Made it a bottom border, was easier with the new wrapper

| ---------------------------------- | ------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

> | <span>client_id</span>             | **Required** | The client_id issued by the VA Lighthouse APIs team team                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |

Reviewer: ```suggestion
| <span>client_id</span>             | **Required** | The client_id issued by the VA API Platform team                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
```

</CodeWrapper>

The response will return a new access_token and refresh_token, if you requested the offline_access scope. 

> [Return to top](#on-this-page)

Reviewer: since the header has a colon, this isn't the right anchor. we may want to explicitly define some ids here for longer headers that we need to have anchors. or, alternatively, we might alter the `slugify` function for the `markdown-it-anchor` (in the Webpack configs) so that it removes colons.

Author: it works when I click on the link but maybe I'm missing something. Changed to #page-header in the wrapper and that seems to be work as well.

## Scopes {tabIndex="-1"}
---
Scopes define the API endpoint your application is allowed to access. We suggest requesting the fewest number of scopes for which you require a Veteran to provide consent. You can always request access to additional scopes if a Veteran or VSO needs the data while using your application.  

**Existing Scopes are:**

| Scope                    | Values and description                                                                                                                   |
| ------------------------ | -----------------------------------------------------------------------------------------------------------------------------------------|
| profile                  | Granted by default, allows access to a user's first and last name and email.                                                             |
| offline_access           | This scope causes the authorization server to provide a refresh token when the [access token](#id-token) is requested.                   |

> | openid                   | An id_token is available in the authorization code flow (response_type = code) token response when the 'openid' scope is used          |

Reviewer: ```suggestion
| openid                   | An id_token is available in the authorization code flow (response_type = code) token response when the 'openid' scope is used.          |
```

[Return to top](#on-this-page)
## ID Token {tabIndex="-1"}
---
Access tokens and id_tokens are [JSON Web Tokens](https://jwt.io/) or JWTs. A JWT consists of three parts: a header, a payload, and a signature. 

An id_token is available in the authorization code flow (response_type = code) token response when the openid scope is used.

Your application must validate JWT signatures. This allows your application to verify that the provided JWT originates from our authorization servers and prevents your application from accepting a JWT with claims that are attempting to impersonate one of your users.

### Header

> The JWT's header has two fields alg and kid. alg indicates the algorithm that was used the sign the JWT, and kid identifies the key that was used to sign the JWT. Signing keys and associated metadata are accessible from 

Reviewer: ```suggestion
The JWT's header has two fields, alg and kid. alg indicates the algorithm that was used the sign the JWT, and kid identifies the key that was used to sign the JWT. Signing keys and associated metadata are accessible from 
```

https://sandbox-api.va.gov/oauth2/.well-known/openid-configuration.

### Signature

The signature is a cryptographically generated signature of the JWT's header and payload used to confirm the JWT's authenticity. Your application must validate this signature using the alg and the kid from the JWT's header. You may want use one of the JWT libraries listed at [jwt.io](https://jwt.io) to help make this process easier.

### Payload

The payload is a JSON object containing identity and authentication-related claims. There are a couple claims in the JWT that are important for your application to consider:
- nonce - should match the nonce you initiated authorization with.
- exp - the expiration time of the JWT. The token cannot be accepted by the Lighthouse platform after this time, and your application should not use an expired token to identify a user.

[Return to top](#on-this-page)
## Test Users {tabIndex="-1"}
---

> Some APIs require test users and test data. Most of the test data provided from the Lighthouse platform comes from internal VA systems, is not real data, and are reset based upon new recordings of underlying services. We provide test accounts for you to use while developing your Lighthouse-based application, and you must be able to log in to access this data. These test accounts are API-specific, and contain data that is geared toward each API. 

Reviewer: need the link to the Github repo to follow here

> ### HTTPS

Reviewer: not a q for you, but i don't get why the HTTPS section would be nested under Test Users. should check with Brent and Darcy
